File: 프로그래머스/lv2/12924. 숫자의 표현/숫자의 표현.java
class Solution {
    public int solution(int n) {
        int answer = 0;
        int i=1;
        
        while(i <= n/2){
            int start = i;
            int sum = 0;

            while(sum < n){
                sum += start;
                start++;
            }
            if(sum == n){
                answer++;
            }
            i++;
        }
        answer++;
        return answer;
    }
}

File: 프로그래머스/2/12924. 숫자의 표현/숫자의 표현.js
function solution(n) {
    if (n === 1) return 1;
    
    let answer = n % 2 === 0 ? 1 : 2;
    
    for (let i = 1; i < Math.floor(n / 2); i++) {
        let sum = i;
        for (let j = i + 1; j < n; j++) {
            sum += j;
            
            if (sum === n) {
                answer++;
                break;
            }
            
            if (sum > n) break;
        }
    }
    
    return answer;
}